psycopg/psycopg/connection.py

```diff
@@ -95,27 +95,44 @@ def connect(
         timeout = timeout_from_conninfo(params)
         rv = None
         attempts = conninfo_attempts(params)
+        connection_errors: list[tuple[e._NoTracebackException, str]] = []
```

Comment:
You could use `list[tuple[BaseException, str]]` or even `list[tuple[Exception, str]]` if we stop catching `KeyboardInterrupt/CancelledError` here.

Reply:
Agreed, removing those other exceptions would make the types a lot simpler here

Reply:
Should be addressed in [982dda4](https://github.com/psycopg/psycopg/commit/982dda4f5c99832d95e0e447ba6c16eb9122a24a). I've left the `_NO_TRACEBACK` variable since it's used in quite a few places and may still be needed in the future.